wasi/wasi.go

```diff
@@ -396,13 +424,48 @@ func proc_exit(*wasm.HostFunctionCallContext, uint32) {
 	// not implemented yet
 }
 
-func environ_sizes_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
-	// not implemented yet
-	return
+// environ_sizes_get is a WASI API that returns the number of the environment variables and the total buffer size that
+// environ_get API will require to store the value of the environment variables.
+// * environCountPtr: a pointer to an address of uint32 type. The number of the environment variable is written there.
+// * environBufSizePtr: a pointer to an address of uint32 type. The total size of the buffer that the environment variable data requires is written there.
+//
+// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-environ_sizes_get---errno-size-size
+func (w *WASIEnvironment) environ_sizes_get(ctx *wasm.HostFunctionCallContext, environCountPtr uint32, environBufSizePtr uint32) (err Errno) {
+	if !ctx.Memory.PutUint32(environCountPtr, uint32(len(w.environ.nullTerminatedValues))) {
+		return EINVAL
+	}
+	if !ctx.Memory.PutUint32(environBufSizePtr, w.environ.totalBufSize) {
+		return EINVAL
+	}
+
+	return ESUCCESS
 }
 
-func environ_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
-	return
+// environ_get is a WASI API to read the environment variable data.
+// * environPtr:
+//     A pointer to a buffer. environ_get writes multiple *C.char pointers in sequence there. In other words, this is an array of *char in C.
+//     The pointers point to environment variables. Each environment variable is a key/value pair joined with `=`, and it is a null-terminated string.
```

Comment:
if we match signatures with how we set this, we should be able to say "See Environ for notes on syntax`